src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -86,9 +86,11 @@ export function DiscoverMainRoute({
     });
   const [error, setError] = useState<Error>();
   const [loading, setLoading] = useState(true);
-  const [hasESData, setHasESData] = useState(false);
-  const [hasUserDataView, setHasUserDataView] = useState(false);
-  const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
+  const [noDataState, setNoDataState] = useState({
+    hasESData: true,
+    hasUserDataView: true,
+    showNoDataPage: false,
+  });
```

Comment:
A bit of optimization, so let `setState` is being called, this will lead to less re-renderings, however those are just small wins